parser.cpp

```diff
@@ -1746,6 +1756,60 @@ namespace Sass {
     return cond;
   }
 
+  At_Root_Block* Parser::parse_at_root_block()
+  {
+    lex<at_root>();
+    Position at_source_position = source_position;
+    Block* body = 0;
+    At_Root_Expression* expr = 0;
+    Selector_Lookahead lookahead_result;
+    in_at_root = true;
+    if (peek< exactly<'('> >()) {
+      expr = parse_at_root_expression();
+      body = parse_block();
+    }
+    else if (peek< exactly<'{'> >()) {
+      body = parse_block();
+    }
+    else if ((lookahead_result = lookahead_for_selector(position)).found) {
+      Ruleset* r = parse_ruleset(lookahead_result);
+      body = new (ctx.mem) Block(r->path(), r->position(), 1);
+      *body << r;
+    }
+    in_at_root = false;
+    At_Root_Block* at_root = new (ctx.mem) At_Root_Block(path, at_source_position, body);
+    if (expr) at_root->expression(expr);
+    return at_root;
+  }
+
+  At_Root_Expression* Parser::parse_at_root_expression()
```

Comment:
I'm want to experiment with how this looks with the `At_Root_Expression`. I don't think that AST node makes a lot of sense.